### PR TITLE
Added documentation for linear search

### PR DESCRIPTION
**Pull request by SAY-droid427**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

Yes

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Closes https://github.com/TesseractCoding/NeoAlgo-Docs/issues/13

### CODE REVIEW

File: docs/searching-algorithms/linear-search.mdx [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+        if arr[i] == x:
+            pos = i
+            print("Found at {}".format(pos))
+            
+    if pos == -1:
+        print("Not Found")
+
+```
+</TabItem>
+<Tabs>
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:141]: Keep `</Tabs>` here. The CI checks would pass.

File: docs/searching-algorithms/linear-search.mdx [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+In linear search, no extra space is used to store anything. So the space complexity is O(1).
+
+* Time Complexity: O(n)
+* Space Complexity: O(1)
+
+
+
+### Credits
+[Utkarsh Chaurasia](https://github.com/UtkarshChaurasia) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Linear_Search.cpp) and [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/search/Linear_Search.py)
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:174]: `and the Python implementation`

File: docs/searching-algorithms/linear-search.mdx [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+ The key does not match  arr[0]
+
+ Next key is compared to arr[1] = 3
+ Key = 5 does not match the arr[1]
+
+ The end of the array is reached. 
+ The key is not present in the array
+ -1 is returned indicating the absence of the key in array 
+
+```
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:61]: Use a smaller example that is easy to consume. Be minimal :)

File: docs/searching-algorithms/linear-search.mdx [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="js"
+  values={[
+    { label: 'C++', value: 'cp', },
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:71]: It should be `cpp`

File: docs/searching-algorithms/linear-search.mdx [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+
+```
+
+# Code
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+<Tabs
+  defaultValue="js"
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:69]: Keep the `defaultValue` as `cpp`

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/searching-algorithms/linear-search.mdx (hunk 1 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Yeah done

File: docs/searching-algorithms/linear-search.mdx (hunk 2 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 3 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: I have tried to use a simpler example with a smaller explanation

File: docs/searching-algorithms/linear-search.mdx (hunk 4 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 5 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx [hunk 6]
```diff
@@ -0,0 +1,172 @@
+---
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
+
+---
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:9]: It has been repeated twice. Kindly rectify.

File: docs/searching-algorithms/linear-search.mdx [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+
+# Introduction
+
+Linear search is the simplest of the search algorithms. The motive of any searching algorithm is to search for the presence of a key element in the collection of elements( arrays, linked lists, etc). Linear search starts the search of the key element(the element whose presence in the given list is to be searched) sequentially, from the beginning to the list until it is found.
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:13]: Keep it `search of the key element (the element whose presence in the given list is to be searched) sequentially`

File: docs/searching-algorithms/linear-search.mdx [hunk 8]
```diff
@@ -0,0 +1,172 @@
...
+
+# Introduction
+
+Linear search is the simplest of the search algorithms. The motive of any searching algorithm is to search for the presence of a key element in the collection of elements( arrays, linked lists, etc). Linear search starts the search of the key element(the element whose presence in the given list is to be searched) sequentially, from the beginning to the list until it is found. 
+
+# Algorithm
+
+In linear search, we follow the following algorithm:
+
+* Starting from the leftmost element of the array, compare each element with the key element
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:19]: - Replace `leftmost` with `left-most`
- Replace `compare` with `compare with`

File: docs/searching-algorithms/linear-search.mdx [hunk 9]
```diff
@@ -0,0 +1,172 @@
...
+
+* If the element compared is same as the key, then return the index of the element which matched the key
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:21]: - Replace `same as the key` with `same as the key element`

File: docs/searching-algorithms/linear-search.mdx [hunk 10]
```diff
@@ -0,0 +1,172 @@
...
+
+* If the key does not match any element scanned in the array, then return -1
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:23]: Keep `-1` in backticks

File: docs/searching-algorithms/linear-search.mdx [hunk 11]
```diff
@@ -0,0 +1,172 @@
...
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
+
+---
+title: Linear Search
+slug: /searching-algorithms/linear-search
+---
+
+# Introduction
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:11]: Use Heading-2

File: docs/searching-algorithms/linear-search.mdx [hunk 12]
```diff
@@ -0,0 +1,172 @@
...
+
+Linear search is the simplest of the search algorithms. The motive of any searching algorithm is to search for the presence of a key element in the collection of elements( arrays, linked lists, etc). Linear search starts the search of the key element(the element whose presence in the given list is to be searched) sequentially, from the beginning to the list until it is found. 
+
+# Algorithm
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:15]: Use Heading-2

File: docs/searching-algorithms/linear-search.mdx [hunk 13]
```diff
@@ -0,0 +1,172 @@
...
+In linear search, we follow the following algorithm:
+
+* Starting from the leftmost element of the array, compare each element with the key element
+
+* If the element compared is same as the key, then return the index of the element which matched the key
+
+* If the key does not match any element scanned in the array, then return -1
+
+
+# Explanation for Linear Search
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:26]: Use Heading-2

File: docs/searching-algorithms/linear-search.mdx [hunk 14]
```diff
@@ -0,0 +1,172 @@
...
+ The key does not match  arr[0]
+ Next, key is compared to arr[1] = 3
+ Key = 5 does not match the arr[1]
+
+ The end of the array is reached, but the key is not found in the array. Hence it is not present. 
+ -1 is returned.
+
+```
+
+# Code
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:60]: Use Heading-2

File: docs/searching-algorithms/linear-search.mdx [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+
+```cpp
+#include<bits/stdc++.h>
+using namespace std;
+
+//Linear Search Algorithm
+#include<iostream>
+using namespace std;
+
+int main()
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:82]: Can you put all the code logic in a separate function? Use `main()` only for the driver code

File: docs/searching-algorithms/linear-search.mdx [hunk 16]
```diff
@@ -0,0 +1,172 @@
...
+def linearSearch(arr, x):
+    pos = -1
+    for i in range(len(arr)):
+        if arr[i] == x:
+            pos = i
+            print("Found at {}".format(pos))
+            
+    if pos == -1:
+        print("Not Found")
+
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:135]: Add a Driver Code with `if__name__ == "__main__":`

File: docs/searching-algorithms/linear-search.mdx [hunk 17]
```diff
@@ -0,0 +1,172 @@
...
+```
+</TabItem>
+</Tabs>
+
+# Sample Input and Output
+
+
+## Sample 1 for the C++
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:143]: No need of sample only for C++ 

Keep the sample I/O same for both the languages

File: docs/searching-algorithms/linear-search.mdx [hunk 18]
```diff
@@ -0,0 +1,172 @@
...
+
+### Input :
+
+```
+Enter number of elements:5
+Enter the elements 1 2 4 3 5
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:149]: Keep a Colon after elements

File: docs/searching-algorithms/linear-search.mdx [hunk 19]
```diff
@@ -0,0 +1,172 @@
...
+
+# Sample Input and Output
+
+
+## Sample 1 for the C++
+
+### Input :
+
+```
+Enter number of elements:5
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:148]: Keep a space after colon

File: docs/searching-algorithms/linear-search.mdx [hunk 20]
```diff
@@ -0,0 +1,172 @@
...
+
+In linear search, no extra space is used to store anything. So the space complexity is O(1).
+
+* Time Complexity: O(n)
+* Space Complexity: O(1)
+
+
+
+### Credits
+[Utkarsh Chaurasia](https://github.com/UtkarshChaurasia) for the [C++ implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/C-Plus-Plus/search/Linear_Search.cpp) and the [Python implementation](https://github.com/TesseractCoding/NeoAlgo/blob/master/Python/search/Linear_Search.py)
```
Comment (HarshCasper) [inline docs/searching-algorithms/linear-search.mdx:171]: Use a bullet point

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/searching-algorithms/linear-search.mdx (hunk 7 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 8 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 9 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 10 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

**Comment by HarshCasper** [comment]
Hi @SAY-droid427 

Some of the changes requested are not yet resolved. Kindly resolve them and I will give a final review.

File: docs/searching-algorithms/linear-search.mdx (hunk 19 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx:155]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 18 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx:156]: Done

File: docs/searching-algorithms/linear-search.mdx (hunk 17 above)
Comment (SAY-droid427) [reply docs/searching-algorithms/linear-search.mdx]: Done

**Comment by SAY-droid427** [comment]
@HarshCasper I have made all the changes requested

**Review by HarshCasper — APPROVED** [review]

**Pull request merged**